compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/test/ctrlpkt_gen.mlir

```diff
@@ -0,0 +1,460 @@
+      aie.next_bd ^bb7
+    ^bb9:  // pred: ^bb6
+      %3 = aie.dma_start(MM2S, 1, ^bb10, ^bb12)
+    ^bb10:  // 2 preds: ^bb9, ^bb11
+      aie.use_lock(%lock_0_1_1, AcquireGreaterEqual, 1)
+      aie.dma_bd_packet(0, 0)
+      aie.dma_bd(%buffer_0_1 : memref<64xi8>) {bd_id = 26 : i32, len = 64 : i32, next_bd_id = 27 : i32}
+      aie.use_lock(%lock_0_1, Release, 1)
+      aie.next_bd ^bb11
+    ^bb11:  // pred: ^bb10
+      aie.use_lock(%lock_0_1_1, AcquireGreaterEqual, 1)
+      aie.dma_bd_packet(0, 0)
+      aie.dma_bd(%buffer_0_1_0 : memref<64xi8>) {bd_id = 27 : i32, len = 64 : i32, next_bd_id = 26 : i32}
+      aie.use_lock(%lock_0_1, Release, 1)
+      aie.next_bd ^bb10
+    ^bb12:  // pred: ^bb9
+      %4 = aie.dma_start(S2MM, 2, ^bb13, ^bb15)
+    ^bb13:  // 2 preds: ^bb12, ^bb14
+      aie.use_lock(%lock_0_1_8, AcquireGreaterEqual, 1)
+      aie.dma_bd(%buffer_0_1_6 : memref<64xi32>) {bd_id = 4 : i32, len = 64 : i32, next_bd_id = 5 : i32}
+      aie.use_lock(%lock_0_1_9, Release, 1)
+      aie.next_bd ^bb14
+    ^bb14:  // pred: ^bb13
+      aie.use_lock(%lock_0_1_8, AcquireGreaterEqual, 1)
+      aie.dma_bd(%buffer_0_1_7 : memref<64xi32>) {bd_id = 5 : i32, len = 64 : i32, next_bd_id = 4 : i32}
+      aie.use_lock(%lock_0_1_9, Release, 1)
+      aie.next_bd ^bb13
+    ^bb15:  // pred: ^bb12
+      %5 = aie.dma_start(MM2S, 2, ^bb16, ^bb18)
+    ^bb16:  // 2 preds: ^bb15, ^bb17
+      aie.use_lock(%lock_0_1_9, AcquireGreaterEqual, 1)
+      aie.dma_bd_packet(0, 0)
+      aie.dma_bd(%buffer_0_1_6 : memref<64xi32>) {bd_id = 6 : i32, len = 64 : i32, next_bd_id = 7 : i32}
+      aie.use_lock(%lock_0_1_8, Release, 1)
+      aie.next_bd ^bb17
+    ^bb17:  // pred: ^bb16
+      aie.use_lock(%lock_0_1_9, AcquireGreaterEqual, 1)
+      aie.dma_bd_packet(0, 0)
+      aie.dma_bd(%buffer_0_1_7 : memref<64xi32>) {bd_id = 7 : i32, len = 64 : i32, next_bd_id = 6 : i32}
+      aie.use_lock(%lock_0_1_8, Release, 1)
+      aie.next_bd ^bb16
+    ^bb18:  // pred: ^bb15
+      aie.end
+    }
+    aie.shim_dma_allocation @shim_2(S2MM, 0, 0)
+    %switchbox_0_0 = aie.switchbox(%tile_0_0) {
+      aie.connect<CTRL : 0, SOUTH : 0>
+      %0 = aie.amsel<0> (0)
+      %1 = aie.amsel<1> (0)
+      %2 = aie.amsel<2> (0)
+      %3 = aie.amsel<3> (0)
+      %4 = aie.amsel<4> (0)
+      %5 = aie.masterset(CTRL : 0, %0)
+      %6 = aie.masterset(NORTH : 1, %1)
+      %7 = aie.masterset(NORTH : 4, %2)
+      %8 = aie.masterset(SOUTH : 2, %4)
+      %9 = aie.masterset(NORTH : 3, %3)
+      aie.packet_rules(SOUTH : 3) {
+        aie.rule(31, 0, %0) {packet_ids = array<i32: 0>}
+        aie.rule(31, 1, %1) {packet_ids = array<i32: 1>}
+        aie.rule(31, 2, %3) {packet_ids = array<i32: 2>}
+      }
+      aie.packet_rules(SOUTH : 7) {
+        aie.rule(30, 0, %2) {packet_ids = array<i32: 0, 1>}
+      }
+      aie.packet_rules(NORTH : 1) {
+        aie.rule(31, 0, %4) {packet_ids = array<i32: 0>}
+      }
+    }
+    %core_0_2 = aie.core(%tile_0_2) {
```

Comment:
I think the size of the core code can be reduced to simplify the test and improve maintainability, like remove the scalar fill in the beginning and the store at the end.